### DISCUSSION TITLE
Never receive gentle toast notifications even when new updates available

### DISCUSSION THREAD

**Discussion by alanhamlett** [sparkle-project/Sparkle#2419] [Q&A]
### Description of the problem

I can select `Check for updates` in my UI menu and get a Sparkle window showing a new update available. Upgrading works. But, I never receive gentle reminders for new updates. I've tried deleting the `SULastCheckTime` default for my app, but never get any gentle reminders.

### Do you use Sandboxing in your app?
No

### Version of `Sparkle.framework` in the latest version of your app
2.5.0-beta.1

### Version of `Sparkle.framework` in the old version of app that your users have (or N/A)
2.5.0-beta.1

### Sparkle's output from Console.app
```
None, no errors
```

App source code: https://github.com/wakatime/macos-wakatime

**Comment by zorgiepoo** [comment]
I don't see Sparkle used in that project so I don't have any context with what your implementation looks like.

The documentation for gentle reminders however is over here -- https://sparkle-project.org/documentation/gentle-reminders/ . This also includes examples.

When gentle reminders are shown is based on your delegate implementation of those methods, specifically starting at [-[SPUStandardUserDriverDelegate standardUserDriverShouldHandleShowingScheduledUpdate:andInImmediateFocus:]](https://sparkle-project.github.io/documentation/api-reference/Protocols/SPUStandardUserDriverDelegate.html#/c:objc(pl)SPUStandardUserDriverDelegate(im)standardUserDriverShouldHandleShowingScheduledUpdate:andInImmediateFocus:) and [-[SPUStandardUserDriverDelegate standardUserDriverWillHandleShowingUpdate:forUpdate:state:]](https://sparkle-project.github.io/documentation/api-reference/Protocols/SPUStandardUserDriverDelegate.html#/c:objc(pl)SPUStandardUserDriverDelegate(im)standardUserDriverWillHandleShowingUpdate:forUpdate:state:). 

The suggested `immediateFocus` provided to this first method may be `YES` if the app was recently launched. If you want to test when it's `NO`, refer to the `defaults write` sample on the [gentle reminders page](https://sparkle-project.org/documentation/gentle-reminders/) that sets the time such that an update check will trigger some significant time after your app has been launched.

If you don't have any custom delegate implementation whatsoever, at least in the context of a backgrounded app, the "gentle" check then is simply not stealing the user's focus and presenting the update window in the background at least when the app has been running for some time.

**Comment by alanhamlett** [comment]
> I don't see Sparkle used in that project so I don't have any context with what your implementation looks like.

I removed Sparkle from the project yesterday. The last version using Sparkle is:
https://github.com/wakatime/macos-wakatime/tree/v2.3.1

**Reply by zorgiepoo** [reply]
Okay, your project is complicated to build, the appcast seems removed, you didn't provide more context based on my response in this thread, you did not update your post mentioning it was removed to save me time, etc. 

Anyway looking at your previous code, the system is probably working as expected and you just didn't test what you wanted to test. You didn't override [-[SPUStandardUserDriverDelegate standardUserDriverShouldHandleShowingScheduledUpdate:andInImmediateFocus:]](https://sparkle-project.github.io/documentation/api-reference/Protocols/SPUStandardUserDriverDelegate.html#/c:objc(pl)SPUStandardUserDriverDelegate(im)standardUserDriverShouldHandleShowingScheduledUpdate:andInImmediateFocus:) so Sparkle always handled showing the update -- and it prefers to show updates in immediate focus near app launch. You probably didn't test an update scheduled check being triggered a significant amount of time after the app has been started too like I mentioned above with the `defaults write` command on our docs. A user notification doesn't get delivered if the app is already the active one, the dock badge would get reset if the user already noticed the update.
